Undefined symbols on iOS when building 0.45.0 with Expo 52

> After upgrading to `0.45.0` I get build errors on both iOS using Expo SDK 52
> 
> iOS:
> ```
> ❌  Undefined symbols for architecture x86_64
> ┌─ Symbol: _EmbeddedPaymentElementViewCls
> └─ Referenced from: _RCTThirdPartyFabricComponentsProvider in libReact-RCTFabric.a[41](RCTThirdPartyFabricComponentsProvider.o)
> ``` 

 _Originally posted by @gleung93 in [#1732](https://github.com/stripe/stripe-react-native/issues/1732#issuecomment-2847908930)_

Same issue here `"@stripe/stripe-react-native": "^0.45.0"` with `"expo": "~52.0.11"` using New Arch. Targeting iOS 18 physical device.

```
❌  Undefined symbols for architecture arm64
┌─ Symbol: _EmbeddedPaymentElementViewCls
└─ Referenced from: _RCTThirdPartyFabricComponentsProvider in libReact-RCTFabric.a[41](RCTThirdPartyFabricComponentsProvider.o)
```

--- a/ios/NewArch/GooglePayButtonComponentView.mm
+++ b/ios/NewArch/GooglePayButtonComponentView.mm
@@ -1,0 +1,9 @@
+#import <React/RCTFabricComponentsPlugins.h>
+#import <React/RCTViewComponentView.h>
+
+// On older versions of react-native every single spec is linked so we need
+// to include a dummy implementation to avoid linker errors.
+Class<RCTComponentViewProtocol> GooglePayButtonCls(void)
+{
+  return RCTViewComponentView.class;
+}

